server/modules/routing/schemarouter/schemarouter.c

```diff
@@ -4044,6 +4154,29 @@ static void handleError (
         switch (action) {
                 case ERRACT_NEW_CONNECTION:
                 {
+									
```

Comment:
This should be done after the call to rses_begin_locked_router_action to make sure no other threads are accessing it.